### PR TITLE
Test image api through local V1 repo, skip network test.

### LINKED ISSUE #10967: [integration-cli] [--net none] Fix TestPullImageOfficialNames
Fix TestPullImageOfficialNames so it will still be a functional test, but will pass when the test container has --net none.

Use the local registry in the container to pull an image. 

Error output when tests are run with --net none:
```
--- FAIL: TestPullImageOfficialNames (0.32s)
	docker_cli_pull_test.go:120: pulling the 'docker.io/hello-world' image from the registry has failed: exit status 1
	docker_cli_pull_test.go:120: pulling the 'index.docker.io/hello-world' image from the registry has failed: exit status 1
	docker_cli_pull_test.go:120: pulling the 'library/hello-world' image from the registry has failed: exit status 1
	docker_cli_pull_test.go:120: pulling the 'docker.io/library/hello-world' image from the registry has failed: exit status 1
	docker_cli_pull_test.go:120: pulling the 'index.docker.io/library/hello-world' image from the registry has failed: exit status 1

```

### PR DESCRIPTION
**Pull request by brahmaroutu**
Closes #10967
Signed-off-by: Srini Brahmaroutu srbrahma@us.ibm.com

### CODE REVIEW

**Comment by dmcgowan** [comment]
The new local v1 test is only being used for a brand new test. I would focus on skipping the tests on no network without requiring pulling in the v1 registry. For pulling in the v1 registry I think this change will go beyond simple code-review of a straight forward change (checking network) and onto whether the integration cli should pull in a 430MB image for a single of a system which we are trying to deprecate.

**Comment by brahmaroutu** [comment]
@dmcgowan I agree I would split this into two PRs. Please see the updated code that just skips the test.
Running in-house v1 registry for similar test cases would make sense and is important as we  should care about regressions before customers move to V2?  We can have a separate PR for that. 
BTW this code is discussed in PR #11048

File: integration-cli/requirements.go [hunk 1]
```diff
@@ -32,6 +33,13 @@ var (
 		func() bool { return supportsExec },
 		"Test requires 'docker exec' capabilities on the tested daemon.",
 	}
+	Network = TestRequirement{
+		func() bool {
+			_, err := http.Get("http://google.com")
```
Comment (crosbymichael) [inline integration-cli/requirements.go]: I'm pretty sure you are going to leak here because you are not closing the connection

**Comment by brahmaroutu** [comment]
@crosbymichael I am closing the response body now, this should fix the leak.

**Comment by brahmaroutu** [comment]
Please see that this is merged, I rebased. I will follow the similar procedure for #10964, #10966 and #10968

**Comment by dmcgowan** [comment]
LGTM

**Comment by dmp42** [comment]
LGTM
ping @jfrazelle @icecrime

**Comment by icecrime** [comment]
LGTM

**Pull request merged**